File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

namespace raindrops {

std::string convert(int number) {
	std::string ans;
	if (number % 3 == 0)
		ans += "Pling";
	if (number % 5 == 0)
		ans += "Plang";
	if (number % 7 == 0)
		ans += "Plong";
	return ans.size() ? ans : std::to_string(number);
}

}  // namespace raindrops

File: solutions/prolog/raindrops/1/raindrops.pl
convert(N, Result) :-
    (N mod 3 =:= 0 -> T1 = "Pling"; T1 = ""),
    (N mod 5 =:= 0 -> T2 = "Plang"; T2 = ""),
    (N mod 7 =:= 0 -> T3 = "Plong"; T3 = ""),
    (T1 = "" , T2 = "" , T3 = "" -> number_string(N, Result) ; 
        string_concat(T1, T2, PartResult),
        string_concat(PartResult, T3, Result) 
    ).
